cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java

```diff
@@ -85,52 +84,41 @@
   private Metrics mrMetrics;
 
   // this is only visible at configure time, not at runtime
-  private final ETLBatchConfig config;
+  private final BatchPhaseSpec phaseSpec;
 
-  public ETLMapReduce(ETLBatchConfig config) {
-    this.config = config;
+  public ETLMapReduce(BatchPhaseSpec phaseSpec) {
+    this.phaseSpec = phaseSpec;
   }
 
   @Override
   public void configure() {
-    setName(NAME);
-    setDescription("MapReduce Driver for ETL Batch Applications");
-
-    PipelineRegisterer pipelineRegisterer = new PipelineRegisterer(getConfigurer(), "batch");
-
-    PipelinePhase pipeline =
-      pipelineRegisterer.registerPlugins(
-        config, TimePartitionedFileSet.class,
-        FileSetProperties.builder()
-          .setInputFormat(AvroKeyInputFormat.class)
-          .setOutputFormat(AvroKeyOutputFormat.class)
-          .setEnableExploreOnCreate(true)
-          .setSerDe("org.apache.hadoop.hive.serde2.avro.AvroSerDe")
-          .setExploreInputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat")
-          .setExploreOutputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat")
-          .setTableProperty("avro.schema.literal", Constants.ERROR_SCHEMA.toString())
-          .build(), true);
-
-    Resources resources = config.getResources();
-    if (resources != null) {
-      setMapperResources(resources);
+    setName(phaseSpec.getPhaseName());
+
+    StringBuilder description = new StringBuilder("DataFlow MapReduce phase executor. Sources");
+
+    for (String sourceName : phaseSpec.getPhase().getSources()) {
+      description.append(" '").append(sourceName).append("'");
     }
-    Resources driverResources = config.getDriverResources();
-    if (driverResources != null) {
-      setDriverResources(driverResources);
+    description.append(" to sinks");
+
+    for (StageInfo sinkInfo : phaseSpec.getPhase().getStagesOfType(BatchSink.PLUGIN_TYPE)) {
+      description.append(" '").append(sinkInfo.getName()).append("'");
     }
+    setDescription(description.toString());
 
-    // following should never happen
-    Preconditions.checkNotNull(pipeline, "Pipeline is null");
-    Preconditions.checkNotNull(pipeline.getSinks(), "Sinks could not be found in program properties");
-    // empty transform list is created during pipeline register
-    Preconditions.checkNotNull(pipeline.getTransforms());
-    Preconditions.checkNotNull(pipeline.getConnections(), "Connections could not be found in program properties");
+    setMapperResources(phaseSpec.getResources());
+    setDriverResources(phaseSpec.getResources());
+
+    if (phaseSpec.getPhase().getSources().size() != 1) {
+      throw new IllegalArgumentException("Pipeline must contain exactly one source.");
```

Comment:
yeah this will be removed when multi-input is implemented.  Note that with how the planner is implemented today, a full pipeline may have multiple sources, but each phase (mapreduce) will only have a single source.